### PR TITLE
Feature/setup authentication

### PR DESCRIPTION
**Pull request by tanloiit2010**

### CODE REVIEW

**Comment by tanloiit2010** [comment]
hi a @tungphuong, 

I completed implement the authentication feature for both FE and BE sides.
Please help to review and give me your feedback.

Thanks,
Loi Le

File: src/Gateways/CRM.Graph.Gateway/Startup.cs [hunk 1]
```diff
@@ -31,6 +32,15 @@ public Startup(IConfiguration configuration)
         // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
         public void ConfigureServices(IServiceCollection services)
         {
+            services.AddCors(o => o.AddPolicy("AllowAllPolicy", builder =>
```
Comment (tungphuong) [inline src/Gateways/CRM.Graph.Gateway/Startup.cs]: It is fine, but will be better if we can configure both "any" or specific

File: src/Gateways/CRM.Graph.Gateway/Startup.cs [hunk 2]
```diff
@@ -87,5 +103,18 @@ private void GrpcRegister(IServiceCollection services)
...
+        private void RegisterAuth(IServiceCollection services)
+        {
+            var identityUrl = Configuration.GetValue<string>("IdentityUrl");
+
+            services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+                .AddJwtBearer((options) =>
+                {
+                    options.Authority = identityUrl;
+                    options.RequireHttpsMetadata = false;
+                    options.Audience = "account";
```
Comment (tungphuong) [inline src/Gateways/CRM.Graph.Gateway/Startup.cs]: - Need to load from configuration
- Why the audience is account? :)

File: src/Web/CRM.Portal.Web/src/App.tsx [hunk 3]
```diff
@@ -5,15 +5,35 @@ import "./App.css";
 
 import { DefaultLayout } from "./containers";
 import { AppCtxProvider } from "./contexts";
+import { ApolloProvider } from '@apollo/react-hooks';
+import ApolloClient from 'apollo-boost';
+import { AuthService, LoggerService } from "services";
+
+const client = new ApolloClient({
+  uri: 'http://localhost:58134/graphql',
```
Comment (tungphuong) [inline src/Web/CRM.Portal.Web/src/App.tsx]: should not hard code the url

File: src/Gateways/CRM.Graph.Gateway/Types/QueryType.cs [hunk 4]
```diff
@@ -13,6 +13,8 @@ protected override void Configure(IObjectTypeDescriptor descriptor)
 
         private static void RegisterContactResource(IObjectTypeDescriptor descriptor)
         {
+            descriptor.Authorize();
```
Comment (tungphuong) [inline src/Gateways/CRM.Graph.Gateway/Types/QueryType.cs]: Is this Hot chocolate api?

File: src/Web/CRM.Portal.Web/src/App.tsx [hunk 5]
```diff
@@ -5,15 +5,35 @@ import "./App.css";
...
+  request: async (operation) => {
```
Comment (tungphuong) [inline src/Web/CRM.Portal.Web/src/App.tsx]: Need to refactor this method to another place, should not add many things into App component

File: src/Web/CRM.Portal.Web/src/components/contacts/ContactList.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
+
+import React from "react";
+import { Table } from 'reactstrap';
+import { useQuery } from '@apollo/react-hooks';
+import { gql } from "apollo-boost";
+import { BeatLoader } from 'react-spinners';
+import { Alert } from 'reactstrap';
+import { css } from '@emotion/core';
+
+const GET_LEADS = gql`
```
Comment (tungphuong) [inline src/Web/CRM.Portal.Web/src/components/contacts/ContactList.tsx]: Need to manay graphQL query in the separated file; with extension is ".graphql". I will discuss with you then

File: src/Web/CRM.Portal.Web/src/components/contacts/ContactList.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+      <BeatLoader
+      css={override}
+        color={'#123abc'}
+        loading={loading}
+      />
+    </div>
+  );
+
+  if (error) {
+    console.log(error);
```
Comment (tungphuong) [inline src/Web/CRM.Portal.Web/src/components/contacts/ContactList.tsx]: should use debug lib to write to console

File: src/Web/CRM.Portal.Web/src/setupProxy.js [hunk 8]
```diff
@@ -8,4 +8,12 @@ module.exports = function(app) {
       '^/authority': ''
     }
   }));
+
+  app.use(proxy('/graphqlAPI', {
+    target: 'http://localhost:58134/graphql',
```
Comment (tungphuong) [inline src/Web/CRM.Portal.Web/src/setupProxy.js]: graphQL port was changed, it should be 5100

File: src/Gateways/CRM.Graph.Gateway/Startup.cs (hunk 1 above)
Comment (tanloiit2010) [reply src/Gateways/CRM.Graph.Gateway/Startup.cs]: fixed

File: src/Gateways/CRM.Graph.Gateway/Startup.cs (hunk 2 above)
Comment (tanloiit2010) [reply src/Gateways/CRM.Graph.Gateway/Startup.cs]: fixed

File: src/Gateways/CRM.Graph.Gateway/Types/QueryType.cs (hunk 4 above)
Comment (tanloiit2010) [reply src/Gateways/CRM.Graph.Gateway/Types/QueryType.cs]: yes, it is provided by Hot chocolate

**Pull request closed without merging**